### PR TITLE
feat: Allow parametrizable int fields

### PR DESCRIPTION
**Pull request by simster7**
Closes: #3574
Closes: #3646

**WIP and still needs discussion**

### CODE REVIEW

**Comment by simster7** [comment]
Concern with this: it would break our users who use our Go types directly

File: util/parametrizable/int.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
+package parametrizable
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+type Int string
```
Comment (alexec) [inline util/parametrizable/int.go:10]: correct approach - could  `intstr.IntOrString` work?

**Comment by simster7** [comment]
* [ ] `TTLSecondsAfterFinished *int32`
* [ ] `Priority *int32`
* [ ] `PodPriority *int32`
* [ ] `Mode *int32`
* [ ] `CompressionLevel *int32`
* [ ] `Factor int32`
* [ ] `Limit *int32`
* [ ] `Parallelism *int64`
* [ ] `ActiveDeadlineSeconds *int64`
* [ ] `Depth *uint64`
* [ ] `SecondsAfterCompletion *intstr.IntOrString`
* [ ] `SecondsAfterSuccess *intstr.IntOrString`
* [ ] `SecondsAfterFailure *intstr.IntOrString`
* [ ] `Parallelism *intstr.IntOrString`
* [ ] `Priority *intstr.IntOrString`
* [ ] `Default *intstr.IntOrString`
* [ ] `Value *intstr.IntOrString`
* [ ] `Default *intstr.IntOrString`
* [ ] `Count *intstr.IntOrString`
* [ ] `Start *intstr.IntOrString`
* [ ] `End *intstr.IntOrString`

File: util/intstrutil/util.go [hunk 2]
```diff
@@ -23,6 +23,15 @@ func Int(is *intstr.IntOrString) (*int, error) {
 	return &i, nil
 }
 
+func Int32(is *intstr.IntOrString) (*int32, error) {
```
Comment (alexec) [inline util/intstrutil/util.go:26]: I would like to add a new func to the as the standard `intstr.Parse()` creates `intstr.IntOrStr` not `*intstr.IntOrStr` which is what we always need

File: util/intstr/parametrizable.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// These are utility functions when using IntOrString to hold either an Int or an Argo Variable
+func Int(is *intstr.IntOrString) (*int, error) {
+	if is == nil {
+		return nil, nil
+	}
+	if is.Type == intstr.String {
```
Comment (alexec) [inline util/intstr/parametrizable.go:16]: just use `is.IntValue()`?

Comment (alexec) [reply util/intstr/parametrizable.go:16]: my guts say we should just swallow errors in the string and return zero

Comment (simster7) [reply util/intstr/parametrizable.go:16]: How do we tell apart an error from an explicit `0`?

**Pull request merged**